File: reservation-service/complete/src/main/java/com/example/reservation/model/Reservation.java
package com.example.reservation.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "reservations")
public class Reservation {
    @Id
    private String reservationId;
    private LocalDateTime reserveDate;
    private String status;
    private String userId;
    private String email;
    private double amount;
    private String stallId;
    private LocalDateTime reserveConfirmDate;
}

File: admin-portal/src/types/reservation.ts
export interface Reservation {
  reservationId: string;
  userId: string;
  stallId: string | number;
  status: "PENDING" | "CONFIRMED" | "CANCELLED";
  reserveDate?: string | Date;
  reserveConfirmDate?: string | Date;
  reserveCancelDate?: string | Date;
  email?: string;
  amount?: number;
  vendorName?: string;
  stallName?: string;
  businessName?: string;
}

